certification/internal/shell/has_minimal_vulns.go

```diff
@@ -1,16 +1,89 @@
 package shell
 
 import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
 	"github.com/komish/preflight/certification"
-	"github.com/komish/preflight/certification/errors"
+	"github.com/komish/preflight/certification/internal/utils/fileutils"
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	ovalFilename = "rhel-8.oval.xml.bz2"
+	ovalUrl      = "https://www.redhat.com/security/data/oval/v2/RHEL8/"
+	reportFile   = "vuln.html"
+)
+
 type HasMinimalVulnerabilitiesCheck struct{}
 
 func (p *HasMinimalVulnerabilitiesCheck) Validate(image string, logger *logrus.Logger) (bool, error) {
-	return false, errors.ErrFeatureNotImplemented
+
+	ovalFileUrl := fmt.Sprintf("%s%s", ovalUrl, ovalFilename)
+
+	tempdir, err := os.MkdirTemp(os.TempDir(), "oval-*")
+	if err != nil {
+		logger.Error("Unable to create temp dir", err)
+		return false, err
+	}
+	logger.Debugf("Oval file dir: %s", tempdir)
+	defer os.RemoveAll(filepath.Dir(tempdir))
+
+	// TODO claen up the names
+	ovalFilePath := filepath.Join(tempdir, ovalFilename)
+	logger.Debugf("Oval file path: %s", ovalFilePath)
+
+	err = fileutils.DownloadFile(ovalFilePath, ovalFileUrl)
+	if err != nil {
+		logger.Error("Unable to download Oval file", err)
+		return false, err
+	}
+	// get the file name
+	r := regexp.MustCompile(`(?P<filename>.*).bz2`)
+	ovalFilePathDecompressed := filepath.Join(tempdir, r.FindStringSubmatch(ovalFilename)[1])
+
+	err = fileutils.Unzip(ovalFilePath, ovalFilePathDecompressed)
+	if err != nil {
+		logger.Error("Unable to unzip Oval file: ", err)
+		return false, err
+	}
+
+	numOfVulns, err := numberOfVulnerabilities(image, ovalFilePathDecompressed, logger)
+	if err != nil {
+		return false, err
+	}
+
+	logger.Debugf("The number of found vulnerabilities: %d", numOfVulns)
+	if numOfVulns > 0 {
+		return false, nil
```

Comment:
Could be a good idea to print the path to the report if we find vulnerabilities. Not an issue now but might need to consider writing to something external when running as a job in a cluster.

Reply:
very good idea! will do it